### PR TITLE
bind search queries to url for share-ability

### LINKED ISSUE #238: Flight Icons Website: Add search criteria to query params in URL
At the moment is not possible to share one particular icon (or search) with someone else using the micro-website page. This is actually very useful, and will become a very common use case/need. 

I don't know how much is complex to do in the existing Ember app, but would be great if we could update the `window.history` path while the user types, and add the search text in the URL as query parameter. 

In this case, one could simply copy&paste the URL, and share it (or attach it as link somewhere).

### PR DESCRIPTION
**Pull request by Dhaulagiri**
## :pushpin: Summary

Closes #238 

## :hammer_and_wrench: Detailed Description

* Migrated search logic to a controller to allow for binding to query params more easily
* Used ember-concurrency to simplify debounce behavior 
* Added minimal state for when no results are returned
* It would probably be a good idea to add some tests for this?

There are most definitely room for improvements as my octane ember knowledge is pretty limited.

## :camera_flash: Screenshots

query

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: ember-flight-icons/tests/dummy/app/components/ds-icon-grid.hbs [hunk 1]
```diff
@@ -58,5 +58,7 @@
         <span>{{meta.description}}</span>
       </p>
     </li>
+  {{else}}
+    No icons found
```
Comment (amyrlam) [inline ember-flight-icons/tests/dummy/app/components/ds-icon-grid.hbs:62]: 💯

**Comment by amyrlam** [comment]
I'll take a stab at adding a test in a new commit, and see what you think. Can drop the commit if undesired!

**Comment by Dhaulagiri** [comment]
@amyrlam go for it!

**Review by amyrlam — APPROVED** [review]
🔍  +1

sorry for the delay, got sidetracked on something for my trip. made a basic test, g2g if good for you. def feel free to push to the new year also!

**Comment by amyrlam** [comment]
* rebased to fix percy

File: ember-flight-icons/tests/dummy/app/components/ds-icon-grid.hbs (hunk 1 above)
Comment (Dhaulagiri) [reply ember-flight-icons/tests/dummy/app/components/ds-icon-grid.hbs]: added a test for this case too now

File: ember-flight-icons/tests/dummy/app/controllers/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      });
+    } else {
+      return icons;
+    }
+  }
+
+  @restartableTask *searchIcons(query) {
+    yield timeout(DEBOUNCE_MS);
+
+    this.query = query;
```
Comment (Dhaulagiri) [inline ember-flight-icons/tests/dummy/app/controllers/index.js:29]: i did this to make sure the actual query param gets updated based on the search string, but it feels like this shouldn't be necessary if the value is properly bound between the query param and the search input like i was expecting 🤷

**Comment by didoo** [comment]
@Dhaulagiri 👏👏👏

**Pull request merged**